### PR TITLE
Nrf5340 netcore

### PR DESCRIPTION
**Pull request by Sandraak**
Added support for the nrf5340-netcore. This support consists mostly of cfg annotations and some very small adjustments.

### CODE REVIEW

**Comment by diondokter** [comment]
Hey, would love to see this merged. @Sandraak is an intern with us at Tweede golf and she's been working on this under my guidance. She has also tested some of the peripherals to check if everything is working alright and it did.

I think it's good enough to be merged as it is right now.

**Comment by jonas-schievink** [comment]
Taking a look :eyes:

File: nrf-hal-common/src/uicr.rs [hunk 1]
```diff
@@ -31,6 +32,7 @@ impl Uicr {
     /// UICR registers can only be set to `0` bits, additional overrides back to `1` can only be
     /// performed by erasing the UICR registers.
     /// - Sets all registers to 0xFFFF_FFFFu32
+
```
Comment (jonas-schievink) [inline nrf-hal-common/src/uicr.rs:35]:
```suggestion
```

File: nrf-hal-common/src/uicr.rs [hunk 2]
```diff
@@ -6,6 +6,7 @@
 //! - nrf52811: Section 4.5
 //! - nrf52832: Section 14
 //! - nrf52840: Section 4.5
+//!
```
Comment (jonas-schievink) [inline nrf-hal-common/src/uicr.rs:9]:
```suggestion

```

File: nrf5340-net-hal/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Hardware Abstration Layer for the Nordic nRF5340
+
+This crate is a Hardware Abstraction Layer (HAL) for the Nordic nRF5340. It
+wraps the PAC (`nrf5340-pac`) and provides high level wrappers for the chip's
+peripherals.
```
Comment (jonas-schievink) [inline nrf5340-net-hal/README.md:5]: The readme refers to a nonexistent crate. Since most HAL crates don't have an extra readme, the file can probably just be deleted.

File: .gitignore [hunk 4]
```diff
@@ -3,7 +3,6 @@ target/
 [._]*.sw[a-p]
 **/*.rs.bk
 Cargo.lock
-Cargo.toml
```
Comment (jonas-schievink) [inline .gitignore:6]: What's the reason for this change? The root `Cargo.toml` should never be added to git.

**Review by jonas-schievink — COMMENTED** [review]
Please format the code with `cargo fmt` and add the new crate to `xtask/src/lib.rs`

**Comment by Sandraak** [comment]
Thanks for the feedback @jonas-schievink !

File: nrf5340-net-hal/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+[package]
+name = "nrf5340-net-hal"
+version = "0.1.0"
```
Comment (jonas-schievink) [inline nrf5340-net-hal/Cargo.toml:3]: All crate versions are kept in sync, since that simplifies maintenance and usage

```suggestion
version = "0.15.0"
```

This should make CI get a bit further

File: nrf5340-net-hal/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+description = "HAL for nRF5340 net SoC"
+readme = "../README.md"
+
+repository = "https://github.com/nrf-rs/nrf-hal"
+authors = [
+    "Jonathan Pallant (42 Technology) <jonathan.pallant@42technology.com>",
+    "Sascha Wise <me@saschawise.com",
```
Comment (jonas-schievink) [inline nrf5340-net-hal/Cargo.toml:10]: Authors list doesn't look correct. You can either list yourself there, or remove the `authors` field, since it's optional.

**Comment by jonas-schievink** [comment]
It looks like your commit email isn't set to one that GitHub recognizes. If you want your contribution to be attributed correctly, consider adding your commit email to your GitHub profile.
